src/Factories/WebTemplatingFactory.php

```diff
@@ -39,7 +39,12 @@ function ( string $name, array $context = [] ): string {
 			new TwigFunction(
 				'translations',
 				function (): string {
-					return json_encode( $this->translations );
+					$result = json_encode( $this->translations );
+					if ( $result === false ) {
+						throw new \RuntimeException( sprintf( "Failed to get JSON representation of: %s",
+							var_export( $this->translations, true ) ) );
+					}
+					return $result;
```

Comment:
This is a good example of an instance where using arrays is hurting us: `$translations` should not be an array (of `array<string,string>` type by the way), but an object that can take of its (de-)serialization. Because these strings come from actual JSON files and if those can be decoded, they can be encoded again. Testing this Twig function have a lot of setup overhead. I recommend we have a look at this together and see how we can refactor translation collection and serialization. 

If we want to merge this PR quickly without extending the scope (refactoring translations into a class), we could do one of two things:

1)  add a "TODO" in here, e.g. "TODO: Convert $ranslations array to a class that can serialize itself to JSON. then we don't have to check for false for PHPSTan's sake".
2) Use `json_encode($this->translations, JSON_THROW_ON_ERROR)` with the comment "We know the translations come from JSON so we can let the JSON Exception be unchecked because that should never happen"

Reply:
I've now done a followup-PR instead in #2933

Reply:
Done.